src/main/java/com/mongodb/jdbc/MongoSQLFunctions.java

```diff
@@ -35,90 +35,114 @@ public static MongoSQLFunctions getInstance() {
                                         new String[] {null}),
                                 new MongoFunction(
                                         CURRENT_TIMESTAMP,
-                                        "date",
+                                        BsonTypeInfo.BSON_DATE.getBsonName(),
                                         "returns the current date and time.",
-                                        new String[] {""},
+                                        new String[] {},
                                         FunctionCategory.TIME_DATE_FUNC),
                                 new MongoFunction(
                                         CURRENT_TIMESTAMP,
-                                        "date",
+                                        BsonTypeInfo.BSON_DATE.getBsonName(),
                                         "returns the current date and time.",
-                                        new String[] {"int"},
-                                        FunctionCategory.TIME_DATE_FUNC),
+                                        // Timestamp precision
+                                        new String[] {BsonTypeInfo.BSON_INT.getBsonName()}),
                                 /**
                                  * Note EXTRACT supports more than YEAR, MONTH, DAY, HOUR, MINUTE,
                                  * SECOND for the unit. It also supports TIMEZONE_HOUR |
                                  * TIMEZONE_MINUTE.
                                  */
                                 new MongoFunction(
                                         EXTRACT,
-                                        "long",
+                                        BsonTypeInfo.BSON_LONG.getBsonName(),
                                         "returns the value of the specified unit from the provided date.",
-                                        new String[] {"string", "date"},
+                                        new String[] {
+                                            BsonTypeInfo.BSON_STRING.getBsonName(),
+                                            BsonTypeInfo.BSON_DATE.getBsonName()
+                                        },
                                         FunctionCategory.TIME_DATE_FUNC),
                                 new MongoFunction(
                                         "LOWER",
-                                        "string",
+                                        BsonTypeInfo.BSON_STRING.getBsonName(),
                                         "returns the provided string with all characters changed to lowercase.",
-                                        new String[] {"string"}),
+                                        new String[] {BsonTypeInfo.BSON_STRING.getBsonName()}),
                                 new MongoFunction(
                                         NULLIF,
-                                        null,
+                                        BsonTypeInfo.BSON_NULL.getBsonName(),
```

Comment:
ah, good catch. 
null is for parameters or return type which are not "set" and can be anything.
BSON_NULL is specifically for a Null value.
Now, unfortunately as you notice in the end, the differentiation disappears and null is mapped to java.sql.Types 0 ('null').
I'm wondering if I should default null to OTHER instead of NULL in the resultset 🤔 None of the options are totally perfect. We would need a type which maps "ANY" and there is nothing like that in java.sql.Types. OTHER seems to be the closest though.
WDYT?

Reply:
Yes, OTHER does seem like a closer return type to use.  Sounds good.